test/unit/passing_optional_arguments_test.lua

```diff
@@ -0,0 +1,164 @@
+    -- tree index
+    -- ========================== --
+    local task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true)
+    -- default iterator_type for tree index is "ALL"
+    t.assert_equals(task.iterator_type, "ALL")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "ALL"})
+    t.assert_equals(task.iterator_type, "ALL")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "EQ"})
+    t.assert_equals(task.iterator_type, "EQ")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "REQ"})
+    t.assert_equals(task.iterator_type, "REQ")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "GT"})
+    t.assert_equals(task.iterator_type, "GT")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "GE"})
+    t.assert_equals(task.iterator_type, "GE")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "LT"})
+    t.assert_equals(task.iterator_type, "LT")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "LE"})
+    t.assert_equals(task.iterator_type, "LE")
+    task:kill()
+
+    --errors
+    t.assert_error_msg_content_equals("Unknown iterator type 'ERROR'",
+            expirationd.start, "clean_all", g.tree.id, helpers.is_expired_true, {iterator_type = "ERROR"})
+
+    -- ========================== --
+    -- hash index
+    -- ========================== --
+    task = expirationd.start("clean_all", g.hash.id, helpers.is_expired_true)
+    -- default iterator_type for hash index is "GE"
+    t.assert_equals(task.iterator_type, "ALL")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.hash.id, helpers.is_expired_true, {iterator_type = "ALL"})
+    t.assert_equals(task.iterator_type, "ALL")
+    task:kill()
+
+
+    task = expirationd.start("clean_all", g.hash.id, helpers.is_expired_true, {start_key = 1, iterator_type = "EQ"})
+    t.assert_equals(task.iterator_type, "EQ")
+    task:kill()
+
+    task = expirationd.start("clean_all", g.hash.id, helpers.is_expired_true, {iterator_type = "GT"})
+    t.assert_equals(task.iterator_type, "GT")
+    task:kill()
+
+    --errors
+    t.assert_error_msg_content_equals("Index 'primary' (HASH) of space 'hash' (memtx) does not support requested iterator type",
+            expirationd.start, "clean_all", g.hash.id, helpers.is_expired_true, {start_key = 1, iterator_type = 1})
+
+    t.assert_error_msg_content_equals("Index 'primary' (HASH) of space 'hash' (memtx) does not support requested iterator type",
+            expirationd.start, "clean_all", g.hash.id, helpers.is_expired_true, {start_key = 1, iterator_type = "GE"})
+
```

Comment:
I would split this huge testcase for number smaller testcases, one iterator type per testcase. When testcases are small it is much simpler to manage them.